docs/control-structures.rst

```diff
@@ -0,0 +1,259 @@
+    * Sending Ether
+
+Lastly, although the default function receives no arguments, it can still access the ``msg`` object, including:
+
+    * the address of who is interacting with the contract (``msg.sender``)
+    * the amount of ETH sent (``msg.value``)
+    * the gas provided (``msg.gas``).
+
+The `__init__` Function
+-----------------------
+
+``__init__`` is a special initialization function that may only be called at the time of deploying a contract. It can be used to set initial values for storage variables. A common use case is to set an ``owner`` variable with the creator the contract:
+
+.. code-block:: python
+
+    owner: address
+
+    def __init__():
+        self.owner = msg.sender
+
+You cannot call to other contract functions from the initialization function.
+
+.. _function-decorators:
+
+Decorators Reference
+--------------------
+
+All functions must include one :ref:`visibility <function-visibility>` decorator (``@external`` or ``@internal``). The remaining decorators are optional.
+
+=============================== ===========================================================
+Decorator                       Description
+=============================== ===========================================================
+``@external``                   Function can only be called externally
+``@internal``                   Function can only be called within current contract
+``@pure``                       Function does read contract state or environment variables
+``@view``                       Function does not alter contract state
+``@payable``                    Function is able to receive Ether
+``@nonreentrant(<unique_key>)`` Function cannot be called back into during an external call
+=============================== ===========================================================
+
+``if`` statements
+=================
+
+The ``if`` statement is a control flow construct used for conditional execution:
+
+.. code-block:: python
+
+    if CONDITION:
+        ...
+
+``CONDITION`` is a boolean or boolean operation. The boolean is evaluated left-to-right, one expression at a time, until the condition is found to be true or false.  If true, the logic in the body of the ``if`` statement is executed.
+
+Note that unlike Python, Vyper does not allow implicit conversion from non-boolean types within the condition of an ``if`` statement. ``if 1: pass`` will fail to compile with a type mismatch.
+
+You can also include ``elif`` and ``else`` statements, to add more conditional statements and a body that executes when the conditionals are false:
+
+.. code-block:: python
+
+    if CONDITION:
+        ...
+    elif OTHER_CONDITION:
+        ...
+    else:
+        ...
+
+``for`` loops
+=============
+
+The ``for`` statement is a control flow construct used to iterate over a value:
+
+.. code-block:: python
+
+    for i in <ITERABLE>:
+        ...
+
+The iterated value can be an array, or generated from the builtin ``range`` function.
```

Comment:
```suggestion
The iterated value can be a static array, or generated from the builtin ``range`` function.
```

I mean, we can probably iterate over a dynamic array, eventually.... because we know max size!